### PR TITLE
feat: parse and compile params

### LINKED ISSUE #1827: Cannot use parameters after a select
Cannot use parameters after a select.

**example:**

```prql
from i = invoices
filter $1 <= i.date or i.date <= $2
select [
  i.id,
  i.total,
]
filter i.total > $3
```

**output in playground:**
```
Error: 
   ╭─[:7:18]
   │
 7 │ filter i.total > $3
   ·                  ─┬  
   ·                   ╰── Unknown name $3
───╯
```

### PR DESCRIPTION
**Pull request by aljazerzen**
Closes #1827

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -18,6 +18,7 @@
   - Dates, times and timestamps have a stricter parsing rules.
   - `let`, `func`, `prql`, `switch` are now treated as keywords.
   - Float literals without fraction part are not allowed anymore (`1.`).
+- Support for SQL parameters with similar syntax (#1957, @aljazerzen)
```
Comment (max-sixty) [inline CHANGELOG.md:23]:
```suggestion
- Full support for SQL parameters with syntax like `$1` or `$1_foo` (#1957, @aljazerzen)
```

**Comment by max-sixty** [comment]
Great!! This should maybe be a "canonical PR" to demonstrate how to add something to the language. Although it touches a lot of files, it's actually not that complicated...

**Comment by aljazerzen** [comment]
Yeah, this one is not canonical - you have to know the whole compiler...

**Comment by max-sixty** [comment]
> Yeah, this one is not canonical

Yes, I had meant if we had a PR to demonstrate how to implement a feature throughout the repo. But I agree it'd be better if we could have examples that didn't require that!

**Pull request merged**